.generated/Add.ts

```diff
@@ -0,0 +1,415 @@
+  SignerProvider,
+  Address,
+  Token,
+  toApiVals,
+  ContractState,
+  node,
+  binToHex,
+  TestContractResult,
+  InputAsset,
+  Asset,
+  HexString,
+  SignDeployContractTxResult,
+  contractIdFromAddress,
+  fromApiArray,
+  ONE_ALPH,
+  groupOfAddress,
+  fromApiVals,
+  subscribeToEvents,
+  SubscribeOptions,
+  Subscription,
+  EventSubscription,
+} from "@alephium/web3";
+
+export namespace Add {
+  export type Fields = {
+    sub: HexString;
+    result: bigint;
+  };
+
+  export type State = Fields & Omit<ContractState, "fields">;
+
+  export type AddEvent = {
+    blockHash: string;
+    txId: string;
+    x: bigint;
+    y: bigint;
+    eventIndex: number;
+  };
+
+  export type Add1Event = {
+    blockHash: string;
+    txId: string;
+    a: bigint;
+    b: bigint;
+    eventIndex: number;
+  };
+
+  export async function deploy(
+    signer: SignerProvider,
+    initFields: Fields,
+    deployParams?: {
+      initialAttoAlphAmount?: bigint;
+      initialTokenAmounts?: Token[];
+      issueTokenAmount?: bigint;
+      gasAmount?: number;
+      gasPrice?: bigint;
+    }
+  ): Promise<SignDeployContractTxResult & { instance: AddInstance }> {
+    const deployResult = await artifact.deploy(signer, {
+      initialFields: initFields,
+      initialAttoAlphAmount: deployParams?.initialAttoAlphAmount,
+      initialTokenAmounts: deployParams?.initialTokenAmounts,
+      issueTokenAmount: deployParams?.issueTokenAmount,
+      gasAmount: deployParams?.gasAmount,
+      gasPrice: deployParams?.gasPrice,
+    });
+    const instance = at(deployResult.contractAddress);
+    return { instance: instance, ...deployResult };
+  }
+
+  export function at(address: string): AddInstance {
+    return new AddInstance(address);
+  }
+
+  // This is used for testing contract functions
+  export function stateForTest(
+    sub: HexString,
+    result: bigint,
+    asset?: Asset,
+    address?: string
+  ): ContractState {
+    const newAsset = {
+      alphAmount: asset?.alphAmount ?? ONE_ALPH,
+      tokens: asset?.tokens,
+    };
+    return Add.artifact.toState(
+      { sub: sub, result: result },
+      newAsset,
+      address
+    );
+  }
+
+  export async function testAddMethod(
+    args: { array: [bigint, bigint] },
+    initFields: Fields,
+    testParams?: {
+      group?: number;
+      address?: string;
+      initialAsset?: Asset;
+      existingContracts?: ContractState[];
+      inputAssets?: InputAsset[];
+    }
+  ): Promise<
+    Omit<TestContractResult, "returns"> & { returns: [[bigint, bigint]] }
+  > {
+    const initialAsset = {
+      alphAmount: testParams?.initialAsset?.alphAmount ?? ONE_ALPH,
+      tokens: testParams?.initialAsset?.tokens,
+    };
+    const _testParams = {
+      ...testParams,
+      testMethodIndex: 0,
+      testArgs: args,
+      initialFields: initFields,
+      initialAsset: initialAsset,
+    };
+    const testResult = await artifact.testPublicMethod("add", _testParams);
```

Comment:
1. We need to convert the result so that `returns` and `events` have proper types
2. We could also convert the result directly from `node....`, like `deploy` function.

Reply:
For returns, it has been converted to a proper typescript type. For ralph array types such as `[[U256; 3]; 2]`, it will be converted to typescript `[[bigint, bigint, bigint], [bigint, bigint, bigint]]` instead of `bigint[][]`. But since only one value is returned here, the return type should be `[bigint, bigint]` rather than `[[bigint, bigint]]`, will fix this, thanks!

Reply:
I found a problem for convert event type, for example:

```
Contract Foo(bar: Bar) {
  fn foo() -> () {
    bar.bar()
  }
}

Contract Bar() {
  event E(v: U256)
  pub fn bar() -> () {
    emit E(0)
  }
}
```

The event is emitted from contract `Bar` when calling `Foo.foo`, in this case, we can not convert the event type.